test/test_global_qbx.py

```diff
@@ -157,20 +187,30 @@ def check_disk_undisturbed_by_sources(centers_panel, sources_panel):
         nodes = stage1_density_nodes[:, sources_panel.discr_slice]
 
         # =distance(centers of panel 1, panel 2)
-        dist = (
-            la.norm((
-                    all_centers[..., np.newaxis]
-                    - nodes[:, np.newaxis, ...]).T,
-                axis=-1)
-            .min())
+        dist = np.min(
+                la.norm(
+                    (all_centers[..., np.newaxis] - nodes[:, np.newaxis, ...]).T,
+                    axis=-1),
+                axis=0)
 
         # Criterion:
         # A center cannot be closer to another panel than to its originating
         # panel.
 
-        rad = expansion_radii[centers_panel.discr_slice]
-        assert (dist >= rad * (1-expansion_disturbance_tolerance)).all(), \
-                (dist, rad, centers_panel.element_nr, sources_panel.element_nr)
+        rad = (
+                (1 - expansion_disturbance_tolerance)
+                * np.tile(expansion_radii[centers_panel.discr_slice], 2))
```

Comment:
This now checks that the distance from each expansion center to the source panel is larger than its expansion radius.

Before it was checking that the distances from all the expansion centers in the panel are larger. This could fail when the curvature was large-ish so that
* there was a large expansion disk on one side of the element
* a small expansion disk on the other side, that was closer to the source geometry than the radius of the large one.

Does that make any sense?

Reply:
If nothing else, this seems to have fixed all the issues and the tests are now passing!

Reply:
I *think* that makes sense. Could you add comments documenting the shape of the various intermediate arrays? I find "shape-heavy" code hard to read without that.

Reply:
Added some shapes in 8958a5a. Hope that helps.